### PR TITLE
Report more object properties on mouse enter

### LINKED ISSUE #15420: Report cell coordinates when navigating with mouse in Excel
### Is your feature request related to a problem? Please describe.

(Requested by a user via email)

When hovering the mouse over cells in Excel, NVDA reports the contents of a given cell, but not the coordinates.

### Describe the solution you'd like

An option to report cell coordinates when hovering with the mouse.

### Additional context

Magnification users particularly may not have the top or left edge of the spreadsheet (where the visual cell rows and columns) visible when navigating.  And if they do, reading them requires scanning across and up from the current location and back which can be difficult for low vision users.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #15420

### Summary of the issue:
When the mouse option "Report role when mouse enters object" is enabled, NVDA reports the role of objects when the mouse enters them. However, the role is in pretty insufficient in many situations, for example:

- Check boxes: checked state is missing
- Table cells: coordinates are missing

### Description of user facing changes
- Renamed "Report role when mouse enters object" to "Report object when mouse enters it"
- When enabled, additional  properties (such as states, cell coordinates, table info) will be reported

### Description of development approach
Implemented some additional logic to speak objects with reason controlTypes.OutputReason.MOUSE, and made event_mouseMove use speech.speakObject.

### Testing strategy:
- Tested in Excel that cell coordinates are read correctly.

### Known issues with pull request:
I didn't touch the config spec as I think it is a bit useless to do so, just for renaming a setting. Feel free to shout out if you think otherwise.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by CyrilleB79** [comment]
I am a bit concerned by the fact that there seem to be few mouse users active on this GitHub repo.

Could we Cc some known people?
Maybe @k-kolev1985?

I also wonder if a person using the mouse may need so much information than someone who cannot see anything and uses only the keyboard. I understand the need to know the cell's coordinates; but I imagine that a link can be seen more easily by some people.

Note: I am not advocating against this PR. I just would like to double check if it's relevant and for which use case.

**Comment by XLTechie** [comment]
CC @britechguy

**Comment by cary-rowen** [comment]
cc @hwf1324

**Comment by britechguy** [comment]
Does this fix, by any chance, possibly resolve what I was asking about in issue #15397 ?

File: source/gui/settingsDialogs.py [hunk 1]
```diff
@@ -1832,10 +1831,12 @@ def makeSettings(self, settingsSizer):
 
 		# Translators: This is the label for a checkbox in the
 		# mouse settings panel.
-		reportObjectRoleText = _("Report &role when mouse enters object")
-		self.reportObjectRoleCheckBox=sHelper.addItem(wx.CheckBox(self,label=reportObjectRoleText))
-		self.bindHelpEvent("MouseSettingsRole", self.reportObjectRoleCheckBox)
```
Comment (seanbudd) [inline source/gui/settingsDialogs.py:1837]: I think the helpEvent anchor should be preserved.

File: user_docs/en/userGuide.t2t [hunk 2]
```diff
@@ -1957,8 +1957,10 @@ The options are character, word, line and paragraph.
 
 To toggle text unit resolution from anywhere, please assign a custom gesture using the [Input Gestures dialog #InputGestures].
 
-==== Report role when mouse enters object ====[MouseSettingsRole]
```
Comment (seanbudd) [inline user_docs/en/userGuide.t2t:1960]: I think the helpEvent anchor should be preserved here too

File: user_docs/en/changes.t2t [hunk 3]
```diff
@@ -12,6 +12,8 @@ What's New in NVDA
 
 
 == Changes ==
+- The option "Report role when mouse enters object" in NVDA's mouse settings category has been renamed to "Report object when mouse enters it". This option now announces additional relevant information about an object when the mouse enters it, such as states (checked/pressed) or cell coordiantes in table. (#15420)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:15]: Minor grammar fixes, new line between sentences.

```suggestion
- The option "Report role when mouse enters object" in NVDA's mouse settings category has been renamed to "Report object when mouse enters it".
This option now announces additional relevant information about an object when the mouse enters it, such as states (checked/pressed) or cell coordinates in a table. (#15420)
```

File: user_docs/en/userGuide.t2t [hunk 4]
```diff
@@ -1957,8 +1957,10 @@ The options are character, word, line and paragraph.
 
 To toggle text unit resolution from anywhere, please assign a custom gesture using the [Input Gestures dialog #InputGestures].
 
-==== Report role when mouse enters object ====[MouseSettingsRole]
-If this checkbox is checked, NVDA will announce the role (type) of object as the mouse moves inside it.
+==== Report object when mouse enters it ====[MouseSettingsReportObjectPropertiesCheckBox]
+If this checkbox is checked, NVDA will announce information about objects as the mouse moves inside them.
+This includes the role (type) of the object as well as states (checked/pressed), cell coordinates in tables, etc.
+Note that the announcement of some object dedtails might be dependent on how other settings, such as in the [object presentation #ObjectPresentationSettings] or [Document Formatting #DocumentFormattingSettings] categories are set.
```
Comment (seanbudd) [inline user_docs/en/userGuide.t2t:1963]:
```suggestion
Note that the announcement of some object details might be dependent on how other settings are set, such as in the [object presentation #ObjectPresentationSettings] or [Document Formatting #DocumentFormattingSettings] categories.
```

**Review by Qchristensen — APPROVED** [review]
Reads well, good work

**Comment by k-kolev1985** [comment]
Actually, I've asked about the mouse tracking being able to report object states many years ago. 
For me that would be useful.

**Comment by hwf1324** [comment]
> I also wonder if a person using the mouse may need so much information than someone who cannot see anything and uses only the keyboard. I understand the need to know the cell's coordinates; but I imagine that a link can be seen more easily by some people.

My thoughts on this. Reading out attributes such as clickable, links, etc. while in browse mode is sometimes a redundancy for me. So I would switch it off in the document format. This is the same for mouse navigation.

**Comment by hwf1324** [comment]
> Does this fix, by any chance, possibly resolve what I was asking about in issue #15397 ?

Unfortunately, after my testing, it didn't improve #15397.
As I said in that question, I'm afraid this is a result of the accessibility interface used by this browser kernel.

Not sure this can be done without making the behavior of mouse navigation and browsing modes consistent.

**Review by seanbudd — APPROVED** [review]
Thanks @leonardder

**Comment by Adriani90** [comment]
Not sure if this started after this PR, but NVDA reports Text every time when entering a plain text with the mouse. Can this be turned off somehow? Role Text is actually not really useful for the user to hear every time in the browser.

**Comment by leonardder** [comment]
Pretty sure this was already the case before. You're best to create a new issue for this.

**Comment by hwf1324** [comment]
> Not sure if this started after this PR, but NVDA reports Text every time when entering a plain text with the mouse. Can this be turned off somehow? Role Text is actually not really useful for the user to hear every time in the browser.

See #15397

This seems to be caused by the accessibility interface used by Chromium?Firefox does not have this problem.

**Comment by Adriani90** [comment]
@LeonarddeR is there anything that argues against having this enabled by default?

**Comment by CyrilleB79** [comment]
@Adriani90, do you mean having "Report object when mouse enters it" checked by default?

I imagine that some partially sighted people using the mouse can see objects role and property (e.g. checked) and just want the text to be read. I cannot say how many want just the text and how many also want the role and the properties. But it's not obvious to me that the default value of this setting should be changed.

**Pull request merged**